### PR TITLE
[WIP] Added tensorflow builders and resolvers for datatypes tensor and dataframe

### LINKED ISSUE #280: Integration with machine learning frameworks
**Describe your problem**

Part of our summer-2021.

### PR DESCRIPTION
**Pull request by chaitravi-ce**
Signed-off-by: chaitravi-ce <chaitravi007@gmail.com>

What do these changes do?
-------------------------
* Added a Builder and Resolver for vineyard::Tensor which will accept data in form of tf.data.Dataset and store as Tensor object, similarly, resolve back to tf.data.Dataset
* Added a builder for vineyard::Dataframe which converts tf.data.Dataset in form of dictionary data and its corresponding labels. The resolver first converts the vineyard Dataframe to Pandas Dataframe and then to tf.data.Dataset object.

Future Work
-------------------------------
* Adding builders and resolvers for Tables and RecordBatches.
* Implementing a single builder and resolver for all datatypes in tensorflow.

Related issue number
--------------------

Part of #280

### CODE REVIEW

**Review by sighingnow — APPROVED** [review]
Overall LGTM. Please check the format error in CI and fix them, after CI turns green we could get this patch landed.

File: python/vineyard/contrib/ml/tensorflow.py [hunk 1]
```diff
@@ -16,4 +16,136 @@
...
+    meta['__values_-size'] = len(cols)
+    meta['partition_index_row_'] = kw.get('partition_index', [0, 0])[0]
+    meta['partition_index_column_'] = kw.get('partition_index', [0, 0])[1]
+    meta['row_batch_index_'] = kw.get('row_batch_index', 0)
+    print(meta)
+    return client.create_metadata(meta)
+
+def tf_builder(client, value, builder, **kw):
+    for x,y in value.take(1):
+        if type(x) is dict:
```
Comment (sighingnow) [inline python/vineyard/contrib/ml/tensorflow.py:79]: Use `isinstance()` instead.

File: python/vineyard/contrib/ml/tensorflow.py [hunk 2]
```diff
@@ -16,4 +16,136 @@
...
+        else:
+            placement = slice(idx, idx + 1, 1)
+        values = np.expand_dims(np_value, 0)
+        blocks.append(Block(values, placement, ndim=2))
+    if 'index_' in meta:
+        index = resolver.run(obj.member('index_'))
+    else:
+        index = np.arange(index_size)
+    df = pd.DataFrame(BlockManager(blocks, [pd.Index(columns), index]))
+    df = df.copy()
```
Comment (sighingnow) [inline python/vineyard/contrib/ml/tensorflow.py:119]: Why do you need a `copy()` here?

File: python/vineyard/contrib/ml/tensorflow.py [hunk 3]
```diff
@@ -16,4 +16,136 @@
...
+    labels = df.pop('target')
+    return tf.data.Dataset.from_tensor_slices((dict(df),labels))
```
Comment (sighingnow) [inline python/vineyard/contrib/ml/tensorflow.py:121]: Leave a space between `,` and the next parameter `labels` please.

**Review by sighingnow — CHANGES_REQUESTED** [review]

File: python/vineyard/contrib/ml/tensorflow.py (hunk 2 above)
Comment (chaitravi-ce) [reply python/vineyard/contrib/ml/tensorflow.py:119]: It was recommended, I can delete that, it works.

**Review by sighingnow — APPROVED** [review]
Thank you @chaitravi-ce. I will get this one merged then you can go ahead for other todo items in this task.

/cc @andydiwenzhu

**Pull request merged**